tokio-quiche/src/http3/driver/streams.rs

```diff
@@ -92,6 +92,37 @@ impl StreamCtx {
         (ctx, PollSender::new(backward_sender), forward_receiver)
     }
 
+    /// Signal that there was a headers flush attempt that failed due
+    /// to insufficient flow control or congestion control window.
+    pub(crate) fn full_headers_flush_blocked(&mut self) {
+        if self.first_full_headers_flush_fail_time.is_none() {
+            self.audit_stats.set_headers_pending_flush(true);
+            self.first_full_headers_flush_fail_time = Some(Instant::now());
+        }
+    }
+
+    /// Signal that there was a headers flush attempt was successful.
+    pub(crate) fn full_headers_flush_success(&mut self) {
+        self.maybe_update_header_flush_duration(true);
+    }
+
+    /// Signal that the headers flush was aborted due to stream being reset.
+    pub(crate) fn full_headers_flush_aborted(&mut self) {
+        self.maybe_update_header_flush_duration(false);
+    }
+
+    fn maybe_update_header_flush_duration(&mut self, flush_successful: bool) {
+        // Clear headers_pending_flush only if header were successfully flushed.
+        if flush_successful {
+            self.audit_stats.set_headers_pending_flush(false);
+        }
+
+        if let Some(first) = self.first_full_headers_flush_fail_time.take() {
+            self.audit_stats
+                .add_header_flush_duration(Instant::now().duration_since(first));
+        }
```

Comment:
Why do we update flush duration on abort? Do we also perform the flush action when the connection is aborted?

Reply:
oops, you're right.